stan/math/prim/mat/fun/ViennaCL.hpp

```diff
@@ -1,34 +1,44 @@
 #ifndef STAN_MATH_PRIM_MAT_FUN_VIENNACL_HPP
-#define STAN_MATH_PRIM_MAT_FUN_VIENNACL_HPP
+  #define STAN_MATH_PRIM_MAT_FUN_VIENNACL_HPP
 
-#ifndef VIENNACL_WITH_EIGEN
-  #define VIENNACL_WITH_EIGEN 1
-#endif
+  #ifdef STAN_GPU
+    #ifndef VIENNACL_WITH_OPENCL
+      #define VIENNACL_WITH_OPENCL 1
+    #endif
+  #else
+    #define STAN_GPU 0
```

Comment:
This makes me a bit worried since after this code runs, STAN_GPU will be defined... I see where you're using it as part of a conditional but we might not need that (see below) in which case maybe we can just use `#ifdef`